packages/backend/src/singletons/Synchronizer.mjs

```diff
@@ -218,9 +218,13 @@ export class Synchronizer extends EventEmitter {
       return a.logIndex - b.logIndex
     })
 
+    const blockNumbers = []
```

Comment:
Try something like
```js
const blockNumbers = events.reduce((acc, event) => {
  return [
    ...acc.filter(num => num !== event.blockNumber),
    event.blockNumber
  ]
}, [])
```

This will give you a unique list of block numbers